be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -104,12 +109,20 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
 
     LOG(INFO) << "Prepare(): query_id=" << print_id(query_id)
               << " fragment_instance_id=" << print_id(params.fragment_instance_id) << " backend_num=" << backend_num;
+    WorkGroupPtr wg;
+    if (request.__isset.workgroup && request.workgroup.id != WorkGroup::DEFAULT_WG_ID) {
+        wg = std::make_shared<WorkGroup>(request.workgroup);
+        wg = WorkGroupManager::instance()->add_workgroup(wg);
```

Comment:
If user sets the session variable `workgroup_id` as an unknown number, we will create a workgroup in the BE here.
Should we use the default workgroup in the case, or FE will guarantee that the `workgroup_id` must exist?

Reply:
ok

Reply:
workgoup is pigbacked by TExecPlanFragmentParams, it maybe absent. so for the first time that workgroup is delivered to BE, BE create workgroup and register it.